src/parser/core/modules/ActionWindow/windows/RaidBuffWindow.ts

```diff
@@ -0,0 +1,79 @@
+import {Event, Events} from 'event'
+import {BuffWindow, EvaluatedAction, PlayersBuffedEvaluator} from 'parser/core/modules/ActionWindow'
+import {HistoryEntry} from 'parser/core/modules/ActionWindow/History'
+import {ensureArray} from 'utilities'
+import {filter, oneOf, noneOf} from '../../../filter'
+
+const FULL_PARTY = 8
+
+/**
+ * Tracks buffs applied to the party.
+ */
+export abstract class RaidBuffWindow extends BuffWindow {
+
+	/**
+	 * Array of all status apply events on players for the buffs specified.
+	 *
+	 * These might ideally belong in the History, but it would break the
+	 * current paradigm of the ActionWindow.
+	 */
+	protected raidBuffApplications = new Array<Events['statusApply']>()
+
+	/**
+	 * Expected number of people who should get the raid buff every time.
+	 */
+	protected expectedCount = FULL_PARTY
+
+	override initialise() {
+		super.initialise()
+
+		const partyMembers = this.parser.pull.actors
+			.filter(actor => actor.playerControlled)
+			.map(actor => actor.id)
+		const playerOwnedIds = this.parser.pull.actors
+			.filter(actor => (actor.owner === this.parser.actor) || actor === this.parser.actor)
+			.map(actor => actor.id)
+		const statusFilter = filter<Event>()
+			.type('statusApply')
+			.status(oneOf(ensureArray(this.buffStatus).map(s => s.id)))
+
+		this.addEventHook(statusFilter
+			.source(oneOf(playerOwnedIds))
+			.target(oneOf(partyMembers)), this.onRaidBuffApply)
+
+		// Duplicate jobs can override buffs
+		this.addEventHook(statusFilter
+			.source(noneOf(playerOwnedIds))
+			.target(this.parser.actor.id), this.maybeReOpenPreviousWindow)
```

Comment:
Our code style is to prefer single line or multiline, but avoid mixing and matching like this. Either each segment gets it's own line, or you just keep it to one. Single line is self explanatory, this suggestion is how multi would look.
```suggestion
		this.addEventHook(
			statusFilter
				.source(oneOf(playerOwnedIds))
				.target(oneOf(partyMembers)),
			this.onRaidBuffApply
		)

		// Duplicate jobs can override buffs
		this.addEventHook(
			statusFilter
				.source(noneOf(playerOwnedIds))
				.target(this.parser.actor.id),
			this.maybeReOpenPreviousWindow
		)
```